core/src/main/scala/org/apache/spark/memory/StorageMemoryPool.scala

```diff
@@ -110,17 +140,21 @@ private[memory] class StorageMemoryPool(lock: Object) extends MemoryPool(lock) w
    * Try to shrink the size of this storage memory pool by `spaceToFree` bytes. Return the number
    * of bytes removed from the pool's capacity.
    */
-  def shrinkPoolToFreeSpace(spaceToFree: Long): Long = lock.synchronized {
-    // First, shrink the pool by reclaiming free memory:
-    val spaceFreedByReleasingUnusedMemory = math.min(spaceToFree, memoryFree)
-    decrementPoolSize(spaceFreedByReleasingUnusedMemory)
-    val remainingSpaceToFree = spaceToFree - spaceFreedByReleasingUnusedMemory
+  def shrinkPoolToFreeSpace(spaceToFree: Long): Long = {
+    var spaceFreedByReleasingUnusedMemory = 0L
```

Comment:
if you do the `val (x, y) = lock.synchronized { ... }` thing here then you don't need a var